nightingale_dispatcher/src/nightingale_dispatcher/task.py

```diff
@@ -1,10 +1,17 @@
 #!/usr/bin/env python3
+from enum import Enum
 
 
-class Task:
+class TaskCodes(Enum):
```

Comment:
nice! One thing I realized is that python allows you to compare different Enums. So TaskCodes.SUCCESS == PhaseStatus.PhaseComplete is valid and will just return False. Enums are still much more readable.

I can make a follow up PR which will prevent us from comparing different enums. so TaskCodes.SUCCESS == PhaseStatus.PhaseComplete will raise an exception that says "dont compare different enums. it makes no sense".

This should make the failure we saw saturday more clear. It wouldve resulted in a clear exception instead of the silent failure it resulted in.